plugins/secretstores/gdchauth/gdchauth.go

```diff
@@ -0,0 +1,219 @@
+package gdchauth
+
+import (
+	"bytes"
+	"context"
+	"crypto/x509"
+	_ "embed"
+	"encoding/json"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/golang-jwt/jwt/v5"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	common_http "github.com/influxdata/telegraf/plugins/common/http"
+	"github.com/influxdata/telegraf/plugins/secretstores"
+)
+
+//go:embed sample.conf
+var sampleConfig string
+
+func (*GdchAuth) SampleConfig() string {
+	return sampleConfig
+}
+
+// GdchAuth is the main authenticator struct
+type GdchAuth struct {
+	Audience           string          `toml:"audience"`
+	Log                telegraf.Logger `toml:"-"`
+	ServiceAccountFile string          `toml:"service_account_file"`
+	TokenExpiryBuffer  config.Duration `toml:"token_expiry_buffer"`
+	HTTPClientConfig   common_http.HTTPClientConfig
+
+	account *serviceAccountKey
+	client  *http.Client
+	expiry  time.Time
+	token   string
+	sync.Mutex
+}
+
+func (g *GdchAuth) Init() error {
+	if g.ServiceAccountFile == "" {
+		return errors.New("service_account_file is required")
+	}
+
+	if g.Audience == "" {
+		return errors.New("audience is required")
+	}
+
+	keyData, err := os.ReadFile(g.ServiceAccountFile)
+	if err != nil {
+		return fmt.Errorf("failed to read service account file: %w", err)
+	}
+
+	g.account = &serviceAccountKey{}
+	if err := json.Unmarshal(keyData, g.account); err != nil {
+		return fmt.Errorf("failed to parse service account JSON: %w", err)
+	}
+
+	if err := g.parsePrivateKey(); err != nil {
+		return err
+	}
+
+	return g.buildHTTPClient()
+}
+
+// Get retrieves the token. The key is ignored as this secret store only provides one secret.
+func (g *GdchAuth) Get(_ string) ([]byte, error) {
```

Comment:
What about a check if the key equals to `token` and error out if not? As that value is what you propose in the readme?

Reply:
IMHO I would just add this check. It doesn't make much sense to just use any key. Token seems right to me, as that is what you actually are retrieving..

While I prefer not to, I'm also fine with keeping as is, but then make that clear in the README.